packages/typespec-client-generator-core/src/package.ts

```diff
@@ -0,0 +1,660 @@
+    kind: "paging",
+    nextLinkLogicalPath: pagedMetadata?.nextLinkSegments || [],
+    nextLinkOperation: pagedMetadata?.nextLinkOperation
+      ? getSdkServiceOperation<TServiceOperation>(
+        context,
+        pagedMetadata.nextLinkOperation,
+        basic.parameters
+      )
+      : undefined,
+  };
+}
+
+function getSdkLroServiceMethod<TServiceOperation extends SdkServiceOperation>(
+  context: SdkContext<TServiceOperation>,
+  operation: Operation
+): SdkLroServiceMethod<TServiceOperation> {
+  const metadata = getLroMetadata(context.program, operation)!;
+  const basicServiceMethod = getSdkBasicServiceMethod<TServiceOperation>(context, operation);
+  basicServiceMethod.response.responsePath =
+    metadata.logicalPath ??
+    (metadata.envelopeResult !== metadata.logicalResult &&
+      basicServiceMethod.operation.verb === "post"
+      ? ".result"
+      : ".");
+  return {
+    ...basicServiceMethod,
+    kind: "lro",
+    __raw_lro_metadata: metadata,
+  };
+}
+
+function getSdkMethodResponse(
+  operation: Operation,
+  responses: Record<number, SdkHttpResponse>
+): SdkMethodResponse {
+  // TODO: put head as bool here
+  const allResponseBodies: SdkType[] = [];
+  let nonBodyExists = false;
+  const headers: SdkServiceResponseHeader[] = [];
+  for (const response of Object.values(responses)) {
+    headers.push(...response.headers);
+    if (response.type) {
+      allResponseBodies.push(response.type);
+    } else {
+      nonBodyExists = true;
+    }
+  }
+  const nullable = nonBodyExists && allResponseBodies.length > 0;
+  const responseTypes = new Set<string>(allResponseBodies.map((x) => getHashForType(x)));
+  let type: SdkType | undefined = undefined;
+  if (responseTypes.size > 1) {
+    // return union of all the different types
+    type = {
+      __raw: operation,
+      kind: "union",
+      values: allResponseBodies,
+      nullable,
+    };
+  } else if (responseTypes) {
+    type = allResponseBodies[0];
+  }
+  return {
+    kind: "method",
+    type,
+  };
+}
+
+function getSdkServiceResponseAndExceptions<TServiceOperation extends SdkServiceOperation>(
+  context: SdkContext<TServiceOperation>,
+  httpOperation: HttpOperation
+): [Record<number | string, SdkHttpResponse>, Record<number | string, SdkHttpResponse>] {
+  const responses: Record<number | string, SdkHttpResponse> = {};
+  const exceptions: Record<number | string | "*", SdkHttpResponse> = {};
+  for (const response of httpOperation.responses) {
+    const headers: SdkServiceResponseHeader[] = [];
+    let body: Type | undefined;
+    for (const data of response.responses) {
+      for (const header of Object.values(data.headers || [])) {
+        headers.push({
+          __raw: header,
+          description: getDocHelper(context, header).description,
+          details: getDocHelper(context, header).details,
+          serializedName: getHeaderFieldName(context.program, header),
+          type: getClientType(context, header.type),
```

Comment:
Not sure if this is the correct location but all of the header responses have `encode` set to `rfc3339` which breaks some [tests](https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/encode/datetime/main.tsp#L304-L342).

Reply:
oh interesting, that's a good point because our code is only doing `encode` for input headers